### PR TITLE
Mbedtls

### PR DESCRIPTION
**Pull request by izahn**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes https://github.com/conda-forge/julia-feedstock/issues/125

Note that this needs https://github.com/conda-forge/staged-recipes/pull/16113 , once someone reviews and merges that we can proceed to review this PR.

### CODE REVIEW

**Comment by mkitti** [comment]
mbedtls does not exist in conda-forge.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -35,13 +38,14 @@ requirements:
     - libosxunwind  # [osx]
     - libunwind  # [linux]
     - arpack
-    - suitesparse =5.4.0
```
Comment (mkitti) [inline recipe/meta.yaml:38]: Why did you remove the pin?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -35,13 +38,14 @@ requirements:
...
+    - suitesparse
     - pcre2
     - git
     - libutf8proc
     - libnghttp2
     - zlib
     - p7zip
+    - mbedtls
```
Comment (mkitti) [inline recipe/meta.yaml:48]: Which package are you referring to? mbedtls does not exist in conda-forge

File: recipe/meta.yaml (hunk 1 above)
Comment (izahn) [reply recipe/meta.yaml:38]: I used https://github.com/archlinux/svntogit-community/blob/packages/julia/trunk/julia-hardcoded-libs.patch to patch out the hard-coded `mbedtls` version. Since that patch also removes `suitesparse` version hard-coding we don't need to pin anymore.

File: recipe/meta.yaml (hunk 2 above)
Comment (izahn) [reply recipe/meta.yaml:48]: > mbedtls does not exist in conda-forge.

As noted in my PR it will once https://github.com/conda-forge/staged-recipes/pull/16113 is merged.

**Comment by izahn** [comment]
@conda-forge-admin please restart ci

**Comment by mkitti** [comment]
For Mac, we need to resolve the problem described in this comment when using system LibGit2:
https://github.com/conda-forge/julia-feedstock/pull/115#issuecomment-892684339

**Comment by izahn** [comment]
> For Mac, we need to resolve the problem described in this comment when using system LibGit2:
> [#115 (comment)](https://github.com/conda-forge/julia-feedstock/pull/115#issuecomment-892684339)

That thread is pretty long, can you summarize what the problem is?

**Comment by mkitti** [comment]
> That thread is pretty long, can you summarize what the problem is?

I linked to a specific comment in the thread. Here's the gist.

> Changing from USE_SYSTEM_LIBGIT2=0 to USE_SYSTEM_LIBGIT2=1 we introduce an error on Mac OS X.

```
Error During Test at $PREFIX/share/julia/stdlib/v1.6/Downloads/test/runtests.jl:12
  Got exception outside of a @test
  error setting certificate verify locations:  CAfile: /Users/runner/miniforge3/conda-bld/julia_1628019948829/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho while requesting https://httpbingo.julialang.org/base64/SnVsaWEgaXMgZ3JlYXQh
  Stacktrace:
    [1] (::Downloads.var"#9#18"{IOStream, Base.DevNull, Nothing, Vector{Pair{String, String}}, Float64, Nothing, Bool, Bool, String, Int64, Bool, Bool})(easy::Downloads.Curl.Easy)
      @ Downloads ~/miniforge3/conda-bld/julia_1628019948829/work/usr/share/julia/stdlib/v1.6/Downloads/src/Downloads.jl:356
```

File: recipe/meta.yaml (hunk 1 above)
Comment (izahn) [reply recipe/meta.yaml:38]: Yes, we should test to make sure that issue is fixed by this PR before merging.

**Comment by mkitti** [comment]
Essentially we need `USE_SYSTEM_LIBGIT2=1` for all platforms. It currently does not work on OS X because of "error setting certificate verify locations". I'm not sure what that error is about.

**Comment by izahn** [comment]
> Essentially we need `USE_SYSTEM_LIBGIT2=1` for all platforms. It currently does not work on OS X because of "error setting certificate verify locations". I'm not sure what that error is about.

I haven't had any luck solving this one. Does using libgit2 from BinaryBuilder on osx cause any known problems?

**Comment by izahn** [comment]
This PR is ready for review. It fixes https://github.com/conda-forge/julia-feedstock/issues/125 and https://github.com/conda-forge/julia-feedstock/issues/118 by applying a patch adapted from the Arch Linux Julia package and using the system `mbedtls` on linux.

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 3]
```diff
@@ -38,16 +37,4 @@ jobs:
...
-        echo "##vso[task.setVariable variable=ARTIFACT_NAME]$artifact_name"
-        if [ -d build_artifacts ]; then
-          echo "##vso[task.setVariable variable=CONDA_BLD_DIR_EXISTS]true"
-        fi
-    displayName: Check for conda build artifacts
-    condition: succeededOrFailed()
-
-  - publish: build_artifacts/
-    artifact: $(ARTIFACT_NAME)
-    condition: eq(variables.CONDA_BLD_DIR_EXISTS, 'true')
```
Comment (mkitti) [inline .azure-pipelines/azure-pipelines-linux.yml:53]: Why is this removed?

Comment (izahn) [reply .azure-pipelines/azure-pipelines-linux.yml:53]: I disabled `store_build_artifacts` in `conda-forge.yml` because I was concerned about using so much storage for all my CI builds. We can turn it back on if you have a use for it.

**Comment by izahn** [comment]
@conda-forge-admin please rerender

**Comment by izahn** [comment]
@conda-forge-admin please rerender

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 3 above)
Comment (izahn) [reply .azure-pipelines/azure-pipelines-linux.yml]: Done in https://github.com/conda-forge/julia-feedstock/pull/126/commits/6d6f9db2e4535e50bf43ee1ad2b5155d94a62a6b and https://github.com/conda-forge/julia-feedstock/pull/126/commits/49a40f0ea76f262e71c40a7596dc0721f84a9f52

**Comment by mkitti** [comment]
> I also added explicit tests to make sure that this PR successfully addresses both #125 and #118

Where are these explicit tests?

**Comment by izahn** [comment]
> > I also added explicit tests to make sure that this PR successfully addresses both #125 and #118
> 
> Where are these explicit tests?

Thanks for catching that! Somehow they didn't make it in. I pushed them up in https://github.com/conda-forge/julia-feedstock/pull/126/commits/77a8d02deb4fa83b484b4958d26b865298314e3f

**Comment by mkitti** [comment]
Things are looking great. I'm just trying to reduce the changes to the minimum. Currently, I think there are some changes that were reverted from #123. I have unreverted them here:
https://github.com/izahn/julia-feedstock/pull/1

**Comment by izahn** [comment]
The changes in https://github.com/izahn/julia-feedstock/pull/1 are all to files generated by `conda smithy rerender`. What is issue you are trying to address there?

**Comment by mkitti** [comment]
I'm trying to minimize the actual diff against master.

**Comment by mkitti** [comment]
I'm ready to merge this after it either does not modify `.ci_support` versus master or we can justify why this pull request should modify `.ci_support` versus master.

**Comment by izahn** [comment]
I'll back it up and let the bot do the rerender so we can see what happens.

**Comment by izahn** [comment]
https://github.com/conda-forge/julia-feedstock/commit/5b1555216925ebf8cbcf45adc24a253ac6addb0d just changes the recipe, none of the CI or migration stuff. Now we rerender:

@conda-forge-admin please rerender

**Comment by izahn** [comment]
The bot ran `conda smithy rerender` for us in https://github.com/conda-forge/julia-feedstock/pull/126/commits/38207157581fdd929a04a1842bf58c026ef884bc . All perfectly normal and expected. Rerendering is standard part of the conda packaging workflow, changes produced by it do not need to be reverted.

**Comment by ngam** [comment]
> > Essentially we need `USE_SYSTEM_LIBGIT2=1` for all platforms. It currently does not work on OS X because of "error setting certificate verify locations". I'm not sure what that error is about.
> 
> I haven't had any luck solving this one. Does using libgit2 from BinaryBuilder on osx cause any known problems?

@izahn @mkitti we can try to see if the new libgit2 patch can resolve this for macs. I will start a new PR.
